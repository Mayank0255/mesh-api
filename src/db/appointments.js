import { performQuery } from ".";

const getAppointmentQuery = `SELECT
	*
FROM
	appointments
WHERE
	id = $1`;

const createAppointmentQuery = `INSERT INTO appointments (type, date, notes, member_id, building_id, request_id, acuity_id)
	VALUES($1, $2, $3, $4, $5, $6, $7)
RETURNING
	*`;

export async function getAppointment(id) {
	return performQuery(getAppointmentQuery, [id]);
}

export async function createAppointment(appointment) {
	return performQuery(createAppointmentQuery, [
		appointment.type,
		appointment.date,
		appointment.notes,
		appointment.member_id,
		appointment.building_id,
		appointment.request_id,
		appointment.acuity_id
	]);
}
