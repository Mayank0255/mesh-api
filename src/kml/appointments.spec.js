import { performQuery } from "../db";
jest.mock("../db");
import { getAppointmentsKML } from "./appointments";

describe("getAppointmentsKML", () => {
  it("renders KML for the retrieved list of appointments", async () => {
    performQuery.mockResolvedValue([
      { type: "install", date: 946713600000, lat: 91.1442, lng: 133.2211, alt: 300, request_id: 1111 },
      { type: "install", date: 946704500000, lat: 91.1423, lng: 133.2910, alt: 100, request_id: 2222 },
      { type: "install", date: 946722200000, lat: 91.1401, lng: 133.2401, alt: 250, request_id: 3333 },
    ]);

    expect(await getAppointmentsKML()).toMatchSnapshot();
  })
});
