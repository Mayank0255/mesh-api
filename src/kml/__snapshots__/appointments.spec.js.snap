// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getAppointmentsKML renders KML for the retrieved list of appointments 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<kml xmlns=\\"http://www.opengis.net/kml/2.2\\">
    <Document>
        <Style id=\\"install\\">
    <IconStyle>
        <scale>0.75</scale> 
        <Icon>
            <href>https://i.imgur.com/4baif2L.png</href>
        </Icon>
        <hotSpot xunits=\\"fraction\\" yunits=\\"fraction\\" x=\\"0.5\\" y=\\"0.5\\"></hotSpot>
    </IconStyle>
    <LabelStyle>
        <scale>0</scale>
    </LabelStyle>
</Style>,<Style id=\\"support\\">
    <IconStyle>
        <scale>0.75</scale> 
        <Icon>
            <href>https://i.imgur.com/qVRzBlS.png</href>
        </Icon>
        <hotSpot xunits=\\"fraction\\" yunits=\\"fraction\\" x=\\"0.5\\" y=\\"0.5\\"></hotSpot>
    </IconStyle>
    <LabelStyle>
        <scale>0</scale>
    </LabelStyle>
</Style>,<Style id=\\"survey\\">
    <IconStyle>
        <scale>0.75</scale> 
        <Icon>
            <href>https://i.imgur.com/4baif2L.png</href>
        </Icon>
        <hotSpot xunits=\\"fraction\\" yunits=\\"fraction\\" x=\\"0.5\\" y=\\"0.5\\"></hotSpot>
    </IconStyle>
    <LabelStyle>
        <scale>0</scale>
    </LabelStyle>
</Style>,
<Placemark>
    <name>Install - Sat, Jan 1</name>
    <ExtendedData>
        <Data name=\\"Type\\">
    <value>Install</value>
</Data>
        <Data name=\\"Date\\">
    <value>Sat, Jan 1</value>
</Data>
        <Data name=\\"Hour\\">
    <value>8:00 AM</value>
</Data>
        <Data name=\\"Links\\">
    <value><a href=\\"https://support.nycmesh.net/scp/tickets.php?a=search&amp;query=1111\\" style=\\"margin-right: 1rem;\\">Ticket →</a></value>
</Data>
    </ExtendedData>
    <Point>
        <altitudeMode>absolute</altitudeMode>
        <coordinates>133.2211,91.1442,300</coordinates>
    </Point>
    <styleUrl>#install</styleUrl>
</Placemark>,
<Placemark>
    <name>Install - Sat, Jan 1</name>
    <ExtendedData>
        <Data name=\\"Type\\">
    <value>Install</value>
</Data>
        <Data name=\\"Date\\">
    <value>Sat, Jan 1</value>
</Data>
        <Data name=\\"Hour\\">
    <value>5:28 AM</value>
</Data>
        <Data name=\\"Links\\">
    <value><a href=\\"https://support.nycmesh.net/scp/tickets.php?a=search&amp;query=2222\\" style=\\"margin-right: 1rem;\\">Ticket →</a></value>
</Data>
    </ExtendedData>
    <Point>
        <altitudeMode>absolute</altitudeMode>
        <coordinates>133.291,91.1423,100</coordinates>
    </Point>
    <styleUrl>#install</styleUrl>
</Placemark>,
<Placemark>
    <name>Install - Sat, Jan 1</name>
    <ExtendedData>
        <Data name=\\"Type\\">
    <value>Install</value>
</Data>
        <Data name=\\"Date\\">
    <value>Sat, Jan 1</value>
</Data>
        <Data name=\\"Hour\\">
    <value>10:23 AM</value>
</Data>
        <Data name=\\"Links\\">
    <value><a href=\\"https://support.nycmesh.net/scp/tickets.php?a=search&amp;query=3333\\" style=\\"margin-right: 1rem;\\">Ticket →</a></value>
</Data>
    </ExtendedData>
    <Point>
        <altitudeMode>absolute</altitudeMode>
        <coordinates>133.2401,91.1401,250</coordinates>
    </Point>
    <styleUrl>#install</styleUrl>
</Placemark>
    </Document>
</kml>"
`;
